technical-documents/postmortems/2022-02-02-inflation-vuln.md

```diff
@@ -0,0 +1,281 @@
+
+## Detection
+
+Jay Freeman (a.k.a. saurik) reported the bug to us via security@optimism.io. He first attempted to
+report via our Immunefi bounty program, but decided to email us since it does not explicitly include
+our L2-Geth code.
+
+## Recovery
+
+The recovery process was executed by a small subset of the team in a private slack channel. The
+timeline and activities were as follows:
+
+### Timeline (MT)
+
+(Using github handles as identifiers)
+
+- 2022-02-02 0925: smartcontracts receives an e-mail from saurik claiming to have found a critical
+  issue in L2Geth. E-mail was sent to securityoptimism.io.
+- 2022-02-02 X: saurik messaged smartcontracts on Discord to make sure we checked the e-mail since
+  he knew we had a prior problem where security advisories went to spam.
+- 2022-02-02 0950: Huddle begins in #security on Slack.
+- 2022-02-02 1058: tynes and smartcontracts confirm the issue on the huddle.
+- 2022-02-02 1112: mslipper joins the huddle and alerts infrastructure providers that there is a
+  live security issue and that an emergency update will be required.
+- 2022-02-02 1206: tynes cuts the following builds:
+  - Mainnet: `0.5.8_b6f79171`
+  - Kovan: `0.5.9_d4c6d824`
+- 2022-02-20 1230: optimisticben deploys to Kovan and mainnet.
+- 2022-02-02 1321: mslipper gives instructions to infra providers on how to upgrade.
+- 2022-02-02 1450: Infura upgrades both Kovan and mainnet.
+- 2022-02-02 2157: Alchemy upgrades both Kovan and mainnet.
+- 2022-02-02 1609: Quicknode upgrades mainnet.
+- 2022-02-03 0732: Quicknode upgrades Kovan.
+- 2022-02-03 1245: smartcontracts alerts Boba.
+- 2022-02-03 1600: Boba patches mainnet.
+- 2022-02-03 1600: smartcontracts alerts Metis. It’s unclear when they patched mainnet - they did
+  this overnight.
+- 2022-02-04 0917: smartcontracts opens
+  [PR #2146](https://github.com/ethereum-optimism/optimism/pull/2146), which we will use to sneak in
+  the fix without publicly disclosing it.
+- 2022-02-05 1950: mslipper merges the finalized patch into PR #2146 after testing, and cuts the
+  release of l2geth version `0.5.11`.
+
+## How it was fixed
+
+The
+[fix](https://github.com/ethereum-optimism/optimism/pull/2146/files#diff-20d698ae9b1041792b702bf7015d0beb3cca36701495eaa45b0b8f587b9ae286R887-R889)
+itself is only 3 lines long, it ensures that when the `SELFDESTRUCT` operation is called in an
+account, its balance (in OVM_ETH) is also immediately set to zero.
+
+## Lessons Learned
+
+In this section we outline the lessons learned, and how our processes have changed or will change as
+a result. These lessons are the product of an internal retrospective, as well as many informal
+discussions which have occurred since the incident.
+
+### On overly-optimistic code reviews
+
+This bug was (obviously) not caught by our code review process. Naturally we want to understand why
+that is, by looking at the PR that introduced it, and the organization context of the time.
+
+#### The PR
```

Comment:
We disabled a lot of the evm tests because they would break on the old ovm and we never added them back and ran them consistently. This would have been caught if we spent time running those tests against our system, ie https://github.com/ethereum/tests. It would have been a lot of work to get that repo working 100% against our system but we could have tried

Reply:
I agree that this should be a top priority for Bedrock and should be noted here

Reply:
great callout.